### ISSUE TITLE
pass (password-store) fails with new version of gnupg2

### ISSUE THREAD

**Issue by janrito** [Homebrew/homebrew#40348]
pass fails on decryption:

```
Passphrase Assertion failed: (!"setbufferlen failed"), function dialog_input, file pinentry-curses.c, line 773.
```

**Comment by DomT4** [comment]
I can't reproduce here on a clean build. Can you try `brew remove pass && brew install -fs pass` and see if it reproduces for you?

**Comment by mlsimpson** [comment]
I had this same issue after a brew update on June 2.  I completely removed all gnupg + pass recipes, found which out of those updated on June 2 (gnupg2, pass, pinentry), then manually installed the immediately previous commit.  Problem was resolved.

**Comment by mlsimpson** [comment]
brew install https://raw.githubusercontent.com/Homebrew/homebrew/e077705de23d955ff6efa788a29702cf48ba3594/Library/Formula/pinentry.rb
    brew install https://raw.githubusercontent.com/Homebrew/homebrew/77979ca1e55adedfb8b6dd1386582529e071b6a2/Library/Formula/gnupg2.rb
    brew install https://raw.githubusercontent.com/Homebrew/homebrew/20f4ba664dc1739dd2cbbba492fbe376eb68585c/Library/Formula/pass.rb
solved it for me.

**Comment by janrito** [comment]
Tried again on a separate computer. Same error. 

Installing the previous commits, as @mlsimpson suggests, solves the immediate problem.

**Comment by DomT4** [comment]
Any idea if `pinentry` or `gnupg2` was the breaking update?

**Comment by chdiza** [comment]
It's probably `pinentry`, as there've been multiple complaints about various things here and on the gnupg list.

**Comment by DomT4** [comment]
Does this persist with the new `pinentry`?

**Comment by deferraz** [comment]
I've just switched to an older pinetry version and my pass is working again

**Comment by janrito** [comment]
`pinentry==0.9.4` seems to have fixed the issue

**Comment by nfelt** [comment]
I also had what I assume was the same issue, though in `pass` this manifested itself as the error `gpg: problem with the agent: End of file` as soon as I tried to start typing into the pinentry dialog.  I only saw the `Assertion failed: (!"setbufferlen failed")` version of the error when I tried running pinentry directly (also as soon as I started typing into the dialog).

But anyway, I can confirm that updating pinentry from 0.9.3 to 0.9.4 fixed the issue.
